torchgeo/samplers/utils.py

```diff
@@ -55,17 +55,16 @@ def get_random_bounding_box(
     """
     t_size = _to_tuple(size)
 
-    width = (bounds.maxx - bounds.minx - t_size[1]) // res
-    height = (bounds.maxy - bounds.miny - t_size[0]) // res
+    # May be negative if bounding box is smaller than patch size
+    width = (bounds.maxx - bounds.minx - t_size[1]) / res
+    height = (bounds.maxy - bounds.miny - t_size[0]) / res
```

Comment:
This is the width/height in pixel units. It no longer needs to be an integer, float is fine too. We cast to integer elsewhere.